src/components/tabs/ImportModal.tsx

```diff
@@ -119,7 +119,7 @@ export default function ImportModal({
       if (link.includes("chess.com")) {
         pgn = await getChesscomGame(link);
       } else if (link.includes("lichess")) {
-        const gameId = link.split("/")[3];
+        const gameId = link.split("?")[0].split("/").pop() || "";
```

Comment:
Here is a cleaner way to do this:

```suggestion
        const excludedPathParts = ['game', 'export', 'white', 'black'];
        const gameId = new URL(link).pathname.split('/').find(x => x && !excludedPathParts.includes(x))
```